kernel/kernel-api/src/main/java/io/delta/kernel/data/FilteredColumnarBatch.java

```diff
@@ -37,10 +39,30 @@
 public class FilteredColumnarBatch {
   private final ColumnarBatch data;
   private final Optional<ColumnVector> selectionVector;
+  private final Optional<String> filePath;
```

Comment:
Can we consider not coupling filePath to ColumnBatch here? FilteredColumnarBatch seems to be a very generic class